lib/redis_memo.rb

```diff
@@ -94,6 +96,29 @@ def self.without_memo
     Thread.current[THREAD_KEY_WITHOUT_MEMO] = prev_value
   end
 
+  def self.with_max_connection_attempts(max_attempts)
+    prev_value = Thread.current[THREAD_KEY_WITHOUT_MEMO]
+    if max_attempts
+      Thread.current[THREAD_KEY_CONNECTION_ATTEMPTS_COUNT] = 0
+      Thread.current[THREAD_KEY_MAX_CONNECTION_ATTEMPTS] = max_attempts
+    end
+    yield
+  ensure
+    Thread.current[THREAD_KEY_WITHOUT_MEMO] = prev_value
+    Thread.current[THREAD_KEY_CONNECTION_ATTEMPTS_COUNT] = nil
+    Thread.current[THREAD_KEY_MAX_CONNECTION_ATTEMPTS] = nil
+  end
+
+  # Fall back to RedisMemo.without_memo if the maximum connection attempts has been reached for a single request
+  def self.incr_max_connection_attempts
```

Comment:
```suggestion
  private
  def self.incr_connection_attempts # :nodoc:
```

Reply:
Is this supposed to be a private method? if so, we can add `# :nodoc:` to skip documentation